src/datasource.ts

```diff
@@ -168,61 +168,70 @@ export default class CogniteDatasource {
     const dataQueryRequestItems = await Promise.all(dataQueryRequestPromises);
 
     const queries: DataQueryRequest[] = [];
-    for (const { target, queryList } of dataQueryRequestItems.map((ql, i) => ({
+    for (let { target, queryList } of dataQueryRequestItems.map((ql, i) => ({
       target: queryTargets[i],
       queryList: ql,
     }))) {
       if (queryList.length === 0 || target.error) {
         continue;
       }
-      // keep track of target lengths so we can assign errors later
-      targetQueriesCount.push({
-        refId: target.refId,
-        count: queryList.length,
-      });
-      // create query requests
-      const queryReq: DataQueryRequest = {
-        items: queryList,
-        start: timeFrom,
-        end: timeTo,
-      };
-      if (target.aggregation && target.aggregation !== 'none') {
-        queryReq.aggregates = target.aggregation;
-        if (!target.granularity) {
-          queryReq.granularity = this.intervalToGranularity(options.intervalMs);
-        } else {
-          queryReq.granularity = target.granularity;
-        }
-      }
-      if (target.assetQuery && target.assetQuery.func && target.tab === Tab.Custom) {
-        if (queryReq.aggregates) {
-          target.error =
-            '[ERROR] To use aggregations with functions, use [ID,aggregation,granularity] or [ID,aggregation]';
-          targetQueriesCount.pop();
-          continue;
-        }
-        let ids = 0;
-        const idRegex = /\[.*?\]/g; // look for [something]
-        for (const q of queryList) {
-          const matches = q.function.match(idRegex);
-          if (!matches) break;
-          const idsObj = {};
-          for (const match of matches) {
-            idsObj[match.substr(1, match.length - 2)] = true;
+
+      // /dataquery is limited to 100 items, so we need to add new calls if we go over 100 items
+      while (queryList.length > 0) {
```

Comment:
lodash has a chunk function, which will split the querylist into objects of `n`. Might be easier to debug than to constantly mutate the running object.

Reply:
yeah that does sound simpler, I'll fix that